services/bettersocial/user/BetterSocialBlockUserV2.js

```diff
@@ -58,15 +136,27 @@ const BetterSocialBlockUserV2 = async (token, selfUserId, targetUserId, source,
         {transaction: t, message, postId, reason}
       );
 
-      if (targetAnonymousUserId?.user_id)
-        await UserBlockUserFunction.userBlock(
-          UserBlockedUser,
-          UserBlockedUserHistory,
+      if (targetAnonymousUserId?.user_id) {
+        let unfollowParams = {
+          token,
+          selfAnonymousUserId,
+          targetAnonymousUserId,
+          UserFollowUser,
+          UserFollowUserHistory,
+          targetUserId,
           selfUserId,
-          targetAnonymousUserId?.user_id,
-          source,
-          {transaction: t, message, postId: '', reason, isAnonymous: true}
-        );
+          t,
+          postId
+        };
+        // Destroy follow from sign A to Sign B
+        await sign_a_unfollow_sign_b(unfollowParams);
+        // Destroy follow from sign B to sign A
+        await sign_b_unfollow_sign_a(unfollowParams);
+        // Destroy follow from sign A to Anon B
+        await sign_a_unfollow_anon_b(unfollowParams);
+        // Destroy follow from sign B to Anon A
+        await sign_b_unfollow_anon_a(unfollowParams);
+      }
     });
   } catch (e) {
     console.log('Error in block user v2 sql transaction');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [129-159]

In this block, the results of the async functions `sign_a_unfollow_sign_b`, `sign_b_unfollow_sign_a`, `sign_a_unfollow_anon_b`, and `sign_b_unfollow_anon_a` are not being checked. If any of these functions fail, the error will be logged but the execution will continue, which might lead to inconsistent data state. Consider checking the result of these function calls and handle the failure case appropriately.



<!-- This is an auto-generated comment by OSS CodeRabbit -->